### PR TITLE
make comment matching more strict

### PR DESCRIPTION
**Pull request by soberstadt**

### CODE REVIEW

**Comment by soberstadt** [comment]
I guess JS doesn't have a "start of string" regex matcher, only `^`

File: src/bot.js [hunk 1]
```diff
@@ -37,7 +37,7 @@ module.exports = app => {
 
   app.on(['issue_comment.created', 'issue_comment.edited'], async context => {
     const message = context.payload.comment.body
-    if (message.match(/merge to stag((ing)|e)/i)) {
+    if (message.match(/^merge to stag((ing)|e)$/i)) {
```
Comment (knutsenm) [inline src/bot.js:40]:
```suggestion
    if (message.match(/^@some-app-name\s.*merge to stag((ing)|e)/i)) {
```

...require bot to be addressed by name, like dependabot does

Comment (soberstadt) [reply src/bot.js:40]: I don't think you can mention a bot. The way that dependabot does this is that you @ mention `dependabot` which is their organization. I don't think we want to @cruglobal to trigger a merge.

See https://github.com/CruGlobal/missionhub-api/pull/1463#issuecomment-645583906 for example.

Comment (knutsenm) [reply src/bot.js:40]: It could be a fake mention, just so that "@pattern" text matches

Comment (soberstadt) [reply src/bot.js:40]: It could :)

**Pull request merged**